packages/node_modules/@ciscospark/internal-plugin-wdm/src/device/device.js

```diff
@@ -141,8 +141,17 @@ const Device = SparkPlugin.extend({
         return uri;
       })
       // it's likely we fail here because we've cycled though all hosts,
-      // now we rely on wdm service to return new available datacenters
-      .catch(() => this._fetchNewUrls(url));
+      // reset all hosts and then retry connecting
+      .catch(() => this._resetAllAndRetry(url));
+  },
+
+  _resetAllAndRetry(url) {
+    if (!url) {
+      return Promise.reject(new Error('`url` is a required parameter'));
+    }
+
+    this.logger.info(`device: Reset the available hosts if we are done trying all the host and then retry ${url}`);
+    return this.serviceCatalog._resetAllAndRetry(url);
   },
 
   _fetchNewUrls(url) {
```

Comment:
Let's remove the prefix `_` and write a header comment because we are exposing this beyond the device file.